### PR TITLE
posix: pread & pwrite

### PR DESCRIPTION
**Pull request by marcinslusarz**
One of the goals of this patchset is decoupling of file handle updates from actual read & write.
Thanks to that it's now possible to pread from one file handle from multiple threads.

Builds on top of PR #101.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/102)

### CODE REVIEW

File: src/libpmemfile-posix/data.c [hunk 1]
```diff
@@ -568,8 +571,13 @@ file_write(PMEMfilepool *pfp, PMEMfile *file, struct pmemfile_inode *inode,
...
 
-	iterate_on_file_range(pfp, file, file->offset, count, (char *)buf,
-	    write_to_blocks);
+	struct pmemfile_block *block = file_find_block(file, file->offset);
+
+	block = iterate_on_file_range(pfp, file->vinode, block, file->offset,
+			count, (char *)buf, write_to_blocks);
+
+	if (block)
+		file->block_pointer_cache = block;
```
Comment (GBuella) [inline src/libpmemfile-posix/data.c]: This sets the `block_pointer_cache` to the last block of the range, doesn't it?
I think, to be more useful for the next `iterate_on_file_range` call, the code should rather remember the first block of the range.

File: tests/posix/mt/mt.cpp [hunk 2]
```diff
@@ -38,16 +38,33 @@
...
 class mt : public pmemfile_test {
 public:
-	mt() : pmemfile_test()
+	long ncpus;
+	pthread_t *threads;
+
+	mt() : pmemfile_test(256 << 20)
 	{
+		ncpus = sysconf(_SC_NPROCESSORS_ONLN);
+		threads = new pthread_t[(size_t)ncpus * 2];
```
Comment (GBuella) [inline tests/posix/mt/mt.cpp]: Could we use C++ standard library threads in tests instead?
There is also http://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency
That would make windows porting far easier as well.

Comment (marcinslusarz) [inline tests/posix/mt/mt.cpp]: Great idea. But I'd like to do it as a separate PR (probably next week).
